spec.emu

```diff
@@ -115,16 +114,24 @@ copyright: false
               1. If _mode_ is ~shortest~, then
                 1. Return ? IteratorCloseAll(_openIters_, NormalCompletion(*undefined*)).
               1. Else if _mode_ is ~strict~, then
-                1. If _observedStrictEnding_ is *false* and _i_ ≠ 0, then
+                1. If _i_ ≠ 0, then
                   1. Return ? IteratorCloseAll(_openIters_, ThrowCompletion(a newly created *TypeError* object)).
-                1. Set _observedStrictEnding_ to *true*.
+                1. Else,
+                  1. For each integer _k_ such that 1 ≤ k ≤ _iterCount_, in ascending order, do
+                    1. Let _open_ be Completion(IteratorStep(_iters_[_k_])).
+                    1. If _result_ is an abrupt completion, then
+                      1. Remove _iters_[_k_] from _openIters_.
+                      1. Return ? IteratorCloseAll(_openIters_, _result_).
+                    1. If _open_ is *false*, then
+                      1. Remove _iters_[_k_] from _openIters_.
+                    1. Else,
+                      1. Return ? IteratorCloseAll(_openIters_, ThrowCompletion(a newly created *TypeError* object)).
+                1. Return ? IteratorCloseAll(_openIters_, NormalCompletion(*undefined*)).
               1. Else,
                 1. Assert: _mode_ is ~longest~.
+                1. If _openIters_ is empty, return ? IteratorCloseAll(_openIters_, NormalCompletion(*undefined*)).
```

Comment:
Why would we call `IteratorCloseAll` on an empty list? This is equivalent to just `Return *undefined*.`.

Reply:
Done. Also addressed the same issue in `~strict~` case, which was a little more subtle.